internal/gitinterface/config_test.go

```diff
@@ -3,18 +3,26 @@
 package gitinterface
 
 import (
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
 func TestGetGitConfig(t *testing.T) {
-	tmpDir := t.TempDir()
+	// Uses os.MkdirTemp instead of t.TempDir to allow test to pass on Windows
+	tmpDir, err := os.MkdirTemp(os.TempDir(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	repo := CreateTestGitRepository(t, tmpDir)
 
 	// CreateTestGitRepository sets our test config
 	config, err := repo.GetGitConfig()
 	assert.Nil(t, err)
 	assert.Equal(t, testName, config["user.name"])
 	assert.Equal(t, testEmail, config["user.email"])
+
+	os.Remove(tmpDir)
```

Comment:
Could this be deferred, so that if anything fails, the temp dir is still cleaned up?

Reply:
I wonder if we can move the MkdirTemp call into CreateTestGitRepository and have that return the path for when we need the location of the repo. That way, we don't need to remove multiple.

+1 on deferring, also think it might be worth doing RemoveAll to avoid any timing issues with trying to remove a tmp dir that gets separately cleaned up for whatever reason.

Reply:
Yes, we should be using RemoveAll, since currently if we wrap the `os.Remove` with an error check, it error's out "remove /var/folders/5f/q255tppn5098m696g3fd1b_m0000gn/T/911169112: directory not empty", which means that nothing is being cleared by `os.Remove` currently.